docs/learn/expert-guides/universal-profile/extend-profile-functionalities.md

```diff
@@ -0,0 +1,556 @@
+}
+```
+
+</details>
+
+### Step 3: Deploy the Extension Contract
+
+We will use the abi and bytecode to deploy the extension contract
+
+```js
+const TipMeABI = require('./TipMe.json').abi;
+const TipMeBytecode = require('./TipMe.json').bytecode;
+
+const TipMeFactory = new ethers.ContractFactory(
+  TipMeABI,
+  TipMeBytecode,
+  signer,
+);
+
+const tipMeContract = await TipMeFactory.deploy();
+await tipMeContract.deployTransaction.wait();
+
+const tipMeExtensionAddress = tipMeContract.addres;
+```
+
+### Step 4: Encode Function Selector and Store in Universal Profile
+
+:::info Requirements
+
+The address calling the `setData(..)` function needs to have `ADDEXTENSIONS` and `CHANGEEXTENSIONS` permission otherwise the call will fail. Check the [keyManager guides](../key-manager/get-controller-permissions.md) to learn more about permissions.
+
+:::
+
+We need to encode the function selector of `tipMe(..)` and store it in the Universal Profile with a specific data key according to the [LSP17-ContractExtension](../../../standards/generic-standards/lsp17-contract-extension.md) standard.
+
+#### Prepare data key and data value
+
+```js
+const { ERC725YDataKeys } = require('@lukso/lsp-smart-contracts');
+
+const tipMeInterface = new ethers.utils.Interface(TipMeABI);
+const tipMeFunctionSelector = tipMeInterface.getSighash('tipMe()');
+
+// Define the DataKey for the extension
+const extensionDataKey =
+  ERC725YDataKeys.LSP17.LSP17ExtensionPrefix +
+  tipMeFunctionSelector.substr(2) +
+  '0'.repeat(32);
```

Comment:
might be nice to explain how this key is generated and where the rules are written (in the standard probably)